src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -358,5 +369,175 @@ public ResponseEntity<List<LitePermissionDTO>> retrievePermissionsByUsername(@Pa
+		if (userSettings == null || userSettings.isEmpty()) {
+		    LOGGER.info("No settings for the current user {}", userName);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		List<UserSettingDTO> userSettingsDTO = userSettingMapper.map2DTOList(userSettings);
+		LOGGER.info("Found {} user settings", userSettingsDTO);
+		return ResponseEntity.ok(userSettingsDTO);
+	}
+	
+	/**
+	 * Returns a {@link UserSettingDTO} of userSetting created.
+	 * 
+	 * @param userSettingDTO -  the {@link UserSettingDTO} to insert.
+	 * @return {@link UserSettingDTO} if the userSetting has been created, null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@PostMapping(value = "/users/settings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> newUserSettings(@Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Create a UserSetting");
+		String userName = userSettingDTO.getUser();
+		userSettingDTO.setId(0);
+		final String ADMIN = "admin";
+		if (!userName.equals(SecurityContextHolder.getContext().getAuthentication().getName()) 
+						&& !SecurityContextHolder.getContext().getAuthentication().getName().equals(ADMIN)) {
+		    throw new OHAPIException(new OHExceptionMessage("Not allowed."));
+		}
+		UserSetting userSetting = userSettingMapper.map2Model(userSettingDTO);
+		userSetting.setUser(userName);
+		UserSetting created = userSettingManager.newUserSetting(userSetting);
+		if (created == null) {
+		    LOGGER.info("UserSetting is not created!");
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not created."));
+		}
+		LOGGER.info("UserSetting successfully created!");
+		return ResponseEntity.status(HttpStatus.CREATED).body(userSettingMapper.map2DTO(created));
+	}
+	
+	/**
+	 * Updates an existing {@link UserSettingDTO}.
+	 * 
+	 * @param userSettingDTO - the {@link UserSettingDTO} to update.
+	 * @param id - id of {@link UserSetting} .
+	 * @return {@link UserSettingDTO} if the UserSetting has been updated , null otherwise.
+	 * @throws OHServiceException.
+	 */
+	@PutMapping(value = "/users/settings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> updateUserSettings(@PathVariable(name = "id") int id, @Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Update a UserSetting");
+		String userName = userSettingDTO.getUser();
+		final String ADMIN = "admin";
+		if (userSettingDTO.getId() != 0 && userSettingDTO.getId() != id) {	
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not found."));
+		}
+		if (!userName.equals(SecurityContextHolder.getContext().getAuthentication().getName()) 
+						&& !SecurityContextHolder.getContext().getAuthentication().getName().equals(ADMIN)) {
+		    throw new OHAPIException(new OHExceptionMessage("Not allowed."));
+		}
```

Comment:
IDKW, here an authenticaed user (e.g. "doctor") can still update other users settings and should be avoided, hence the check seems not working

Reply:
ok, the "doctor" can update itself only, but after initial checks it can put whatever, so this paypload is accepted and should not.

```
{
  "id": 3,
  "configName": "dashboard",
  "user": "laboratorist",
  "configValue": "string"
}
```

Reply:
already fix

Reply:
yes, thanks!